### PR TITLE
[fleet] fix elastic-agent setup

### PR DESCRIPTION
**Pull request by simitt**
## What does this PR do?

Since recent changes of running Fleet Server when starting Elastic Agent (https://github.com/elastic/beats/pull/24220) the elastic-agent setup was broken. This PR fixes the setup as far as possible. 

There is one more change that needs to be merged into Kibana https://github.com/elastic/kibana/pull/92422; until this PR is merged, the Elastic Agent enrollment endlessly waits for the policy to be written by Kibana. One should see the following log line in the Fleet server logs:
`{"message":"Waiting on policy with Fleet Server integration: d5204d20-7fee-11eb-9337-e79c33f6692e"}`

## Why is it important?

To be able to run with Elastic Agent again.

### CODE REVIEW

**Comment by kuisathaverat** [comment]
There are a couple of lines too long 

```
[2021-03-08T09:41:51.109Z] scripts/modules/opbeans.py:28:33: W503 line break before binary operator
[2021-03-08T09:41:51.109Z] scripts/modules/elastic_stack.py:705:121: E501 line too long (124 > 120 characters)
[2021-03-08T09:41:51.109Z] scripts/modules/elastic_stack.py:711:121: E501 line too long (124 > 120 characters)
```

File: scripts/modules/elastic_stack.py [hunk 1]
```diff
@@ -696,34 +696,55 @@ def __init__(self, **options):
...
-        # FLEET_SETUP - if  set to 1 fleet setup will be performed
-        # FLEET_TOKEN_NAME - token name for a token to be created
-        # KIBANA_HOST - actual kibana host [http://localhost:5601]
-        # KIBANA_PASSWORD - password for accessing kibana API [changeme]
-        # KIBANA_USERNAME - username for accessing kibana API [elastic]
+        # ---- Preparing Kibana for Fleet
+        # KIBANA_FLEET_SETUP - set to 1 enables this setup
+
+        # ---- Bootstrapping Fleet Server
+        # This bootstraps the Fleet Server to be run by this Elastic Agent. At least one Fleet Server is required in a Fleet
```
Comment (kuisathaverat) [inline scripts/modules/elastic_stack.py:705]:
```suggestion
        # This bootstraps the Fleet Server to be run by this Elastic Agent. 
        # At least one Fleet Server is required in a Fleet
```

File: scripts/modules/elastic_stack.py [hunk 2]
```diff
@@ -696,34 +696,55 @@ def __init__(self, **options):
...
+        # deployment for other Elastic Agent to bootstrap.
+        # FLEET_SERVER_ENABLE - set to 1 enables bootstrapping of Fleet Server (forces FLEET_ENROLL enabled)
+        # FLEET_SERVER_POLICY_NAME - name of policy for the Fleet Server to use for itself [$FLEET_TOKEN_POLICY_NAME]
+
+        # ---- Elastic Agent Fleet Enrollment
+        # This enrolls the Elastic Agent into a Fleet Server. It is also possible to have this create a new enrollment token
```
Comment (kuisathaverat) [inline scripts/modules/elastic_stack.py:711]:
```suggestion
        # This enrolls the Elastic Agent into a Fleet Server. 
        # It is also possible to have this create a new enrollment token
```

File: scripts/modules/elastic_stack.py [hunk 3]
```diff
@@ -696,34 +696,58 @@ def __init__(self, **options):
...
+            es_scheme = "https" if self.options.get("elasticsearch_enable_tls", False) else "http"
+            es_url = es_scheme + "://admin:changeme@" + self.DEFAULT_ELASTICSEARCH_HOST
+        es_parsed_url = urlparse(es_url)
+
         self.environment = {
+            "KIBANA_FLEET_SETUP": "1",
+            "FLEET_SERVER_ENABLE": "1",
             "FLEET_ENROLL": "1",
-            "FLEET_SETUP": "1",
+            "FLEET_SERVER_POLICY_NAME": "Default policy",  # TODO(simitt): make configurable
```
Comment (jalvz) [inline scripts/modules/elastic_stack.py:737]: oh, are you sure it is "Default policy" and not "Default"?. i hope this would not be needed anyways

**Review by jalvz — COMMENTED** [review]
what about FLEET_URL? wasn't that one required (even if pointing to Kibana)?

File: scripts/modules/elastic_stack.py (hunk 3 above)
Comment (simitt) [reply scripts/modules/elastic_stack.py:737]: Yes, it is `Default policy`; and I already put up https://github.com/elastic/beats/pull/24385 to get rid of this config, but no review yet.

**Comment by simitt** [comment]
> what about FLEET_URL? wasn't that one required (even if pointing to Kibana)? 

Not required, it is actually only used if `FLEET_SERVER_ENABLE` is not set (see https://github.com/elastic/beats/pull/24220/files?file-filters%5B%5D=.asciidoc&file-filters%5B%5D=.go#diff-1f8e1e2079524931fa8f3cf121667a25baa2b998557136500d5e7596b46e9187R215)

File: scripts/modules/opbeans.py [hunk 4]
```diff
@@ -24,7 +24,9 @@ def __init__(self, **options):
         self.agent_local_repo = options.get(self.option_name() + "_agent_local_repo")
         self.opbeans_branch = options.get(self.option_name() + "_branch") or ""
         self.opbeans_repo = options.get(self.option_name() + "_repo") or ""
-        self.es_urls = ",".join(self.options.get("opbeans_elasticsearch_urls") or [self.DEFAULT_ELASTICSEARCH_HOSTS])
+        self.es_urls = ",".join(self.options.get("opbeans_elasticsearch_urls")
```
Comment (kuisathaverat) [inline scripts/modules/opbeans.py:27]:
```suggestion
        self.es_urls = ",".join(self.options.get("opbeans_elasticsearch_urls") )
```

File: scripts/modules/opbeans.py [hunk 5]
```diff
@@ -24,7 +24,9 @@ def __init__(self, **options):
         self.agent_local_repo = options.get(self.option_name() + "_agent_local_repo")
         self.opbeans_branch = options.get(self.option_name() + "_branch") or ""
         self.opbeans_repo = options.get(self.option_name() + "_repo") or ""
-        self.es_urls = ",".join(self.options.get("opbeans_elasticsearch_urls") or [self.DEFAULT_ELASTICSEARCH_HOSTS])
+        self.es_urls = ",".join(self.options.get("opbeans_elasticsearch_urls") 
+        if self.es_urls == "":
+            self.es_urls = [self.DEFAULT_ELASTICSEARCH_HOSTS_NO_TLS])
```
Comment (kuisathaverat) [inline scripts/modules/opbeans.py:29]:
```suggestion
            self.es_urls = [self.DEFAULT_ELASTICSEARCH_HOSTS_NO_TLS]
```

File: scripts/modules/opbeans.py (hunk 4 above)
Comment (kuisathaverat) [reply scripts/modules/opbeans.py:27]: there is still a missing parenthesis

**Comment by simitt** [comment]
Sorry for the noise on this PR - the `make lint` command is broken in my env (using the  `venv` from `make venv`)

**Review by jalvz — APPROVED** [review]

**Pull request merged**